flang/lib/Lower/RTBuilder.h

```diff
@@ -166,17 +173,25 @@ constexpr TypeBuilderFunc getModel<bool &>() {
 }
 
 template <>
-constexpr TypeBuilderFunc getModel<float _Complex>() {
+constexpr TypeBuilderFunc getModel<c_float_complex_t>() {
   return [](mlir::MLIRContext *context) -> mlir::Type {
     return fir::CplxType::get(context, sizeof(float));
   };
 }
 template <>
-constexpr TypeBuilderFunc getModel<double _Complex>() {
+constexpr TypeBuilderFunc getModel<c_double_complex_t>() {
   return [](mlir::MLIRContext *context) -> mlir::Type {
     return fir::CplxType::get(context, sizeof(double));
   };
 }
+template <>
+constexpr TypeBuilderFunc getModel<std::complex<float>>() {
+  return [](mlir::MLIRContext *context) -> mlir::Type {
+    auto realTy = getModel<float>()(context);
+    llvm::SmallVector<mlir::Type, 2> tuple{realTy, realTy};
+    return mlir::TupleType::get(tuple, context);
```

Comment:
Should this be the model, or should it be `fir::CplxType`? It seems odd that it's different at the FIR level.

Reply:
Since std::complex<float> is not a fundamental type, I think we cannot promote it to the same fir type as Fortran complex, otherwise we will not be able to make a distinction in code gen to LLVM when it matters.
When flang is able to target X86 32 bits one way or another, we will have to be careful because I do not think `fir::CplxType` will be lowered to a `tuple<float, float>` when that's the return type (following clang magic for float _Complex, it should probably be `i64` to get the right return ABI). std::complex<float> however will always end up being a tuple<float, float> because that's library class.